e2e test failed

Affected PRs: #495 #484, which needs retest after fixing this.
Logs from e2e test:
```
=== RUN   TestTrafficSplit/replace_latest_tag_of_a_revision_with_old_and_give_latest_to_another_revision
Running 'kn service create echo12 --image gcr.io/knative-samples/helloworld-go -n kne2etests2'...
Running 'kn service describe echo12 -o jsonpath={.status.latestCreatedRevisionName} -n kne2etests2'...
Running 'kn service update echo12 --env TARGET=v2 -n kne2etests2'...
Running 'kn service describe echo12 -o jsonpath={.status.latestCreatedRevisionName} -n kne2etests2'...
Running 'kn service update echo12 --tag echo12-cpjww-1=latest -n kne2etests2'...
Running 'kn service update echo12 --untag latest --tag echo12-cpjww-1=old,echo12-mstfw-2=latest -n kne2etests2'...
Running 'kn service describe echo12 -o jsonpath={range .status.traffic[*]}{.tag}{','}{.revisionName}{','}{.percent}{','}{.latestRevision}{'|'}{end} -n kne2etests2'...
Running 'kn service delete echo12 -n kne2etests2'...
Running 'kubectl delete namespace kne2etests2'...
--- FAIL: TestTrafficSplit (338.92s)
    --- PASS: TestTrafficSplit/tag_two_revisions_as_v1_and_v2_and_give_50-50%_share (39.34s)
    --- PASS: TestTrafficSplit/ramp/up_down_a_revision_to_20%_adjusting_other_traffic_to_accommodate (14.25s)
    --- FAIL: TestTrafficSplit/tag_a_revision_as_candidate,_without_otherwise_changing_any_traffic_split (7.74s)
        traffic_split_test.go:368: assertion failed: 
            --- expectedTargets
            +++ formattedActualTargets
              []e2e.TargetFields{
              	{
              		Tag:      "",
            - 		Revision: "echo2-ytdyn-2",
            + 		Revision: "echo2-qjxkx-1",
              		Percent:  100,
              		Latest:   true,
              	},
              	{Tag: "candidate", Revision: "echo2-qjxkx-1"},
              }
            
    --- FAIL: TestTrafficSplit/tag_a_revision_as_candidate,_set_2%_traffic_adjusting_other_traffic_to_accommodate (7.12s)
        traffic_split_test.go:368: assertion failed: 
            --- expectedTargets
            +++ formattedActualTargets
              []e2e.TargetFields{
              	{
              		Tag:      "",
            - 		Revision: "echo3-rxqpk-2",
            + 		Revision: "echo3-yvncn-1",
              		Percent:  98,
              		Latest:   true,
              	},
              	{Tag: "candidate", Revision: "echo3-yvncn-1", Percent: 2},
              }
            
    --- PASS: TestTrafficSplit/update_tag_for_a_revision_from_candidate_to_current,_tag_current_is_present_on_another_revision (16.57s)
    --- PASS: TestTrafficSplit/update_tag_from_testing_to_staging_for_@latest_revision (8.58s)
    --- PASS: TestTrafficSplit/update_tag_from_testing_to_staging_for_a_revision_(non_@latest) (9.47s)
    --- PASS: TestTrafficSplit/remove_a_revision_with_tag_old_from_traffic_block_entirely (10.26s)
    --- PASS: TestTrafficSplit/tag_a_revision_as_stable_and_current_with_50-50%_traffic (7.37s)
    --- FAIL: TestTrafficSplit/revert_all_traffic_to_latest_ready_revision_of_service (6.42s)
        traffic_split_test.go:368: assertion failed: 
            --- expectedTargets
            +++ formattedActualTargets
              []e2e.TargetFields{
              	{
              		Tag:      "",
            - 		Revision: "echo9-jjnkt-2",
            + 		Revision: "echo9-wymdj-1",
              		Percent:  100,
              		Latest:   true,
              	},
              }
            
    --- PASS: TestTrafficSplit/tag_latest_ready_revision_of_service_as_current (6.66s)
    --- PASS: TestTrafficSplit/update_tag_for_a_revision_as_testing_and_assign_all_the_traffic_to_it: (10.60s)
    --- PASS: TestTrafficSplit/replace_latest_tag_of_a_revision_with_old_and_give_latest_to_another_revision (9.97s)
FAIL
FAIL	knative.dev/client/test/e2e	338.954s
Finished run, return code is 1
XML report written to /logs/artifacts/junit_BCtKZlsF.xml
```

--- a/test/e2e/traffic_split_test.go
+++ b/test/e2e/traffic_split_test.go
@@ -21,6 +21,7 @@ import (
 	"strconv"
 	"strings"
 	"testing"
+	"time"
 
 	"gotest.tools/assert"
 	"knative.dev/client/pkg/util"
@@ -366,6 +367,7 @@ func TestTrafficSplit(t *testing.T) {
 }
 
 func (test *e2eTest) verifyTargets(t *testing.T, serviceName string, expectedTargets []TargetFields) {
+	time.Sleep(3 * time.Second)
 	out := test.serviceDescribeWithJsonPath(t, serviceName, targetsJsonPath)
 	assert.Check(t, out != "")
 	actualTargets, err := splitTargets(out, targetsSeparator, len(expectedTargets))

--- a/test/e2e/traffic_split_test.go
+++ b/test/e2e/traffic_split_test.go
@@ -33,8 +33,8 @@ var targetFieldsLength = 4
 // returns deployed service targets separated by '|' and each target fields seprated by comma
 var targetsJsonPath = "jsonpath={range .status.traffic[*]}{.tag}{','}{.revisionName}{','}{.percent}{','}{.latestRevision}{'|'}{end}"
 
-// returns deployed service latest revision name
-var latestRevisionJsonPath = "jsonpath={.status.latestCreatedRevisionName}"
+// returns latest ready revision name jsonpath
+//var LRRJsonPath = "jsonpath={.status.latestReadyRevisionName}"
 
 // TargetFields are used in e2e to store expected fields per traffic target
 // and actual traffic targets fields of deployed service are converted into struct before comparing
@@ -87,10 +87,12 @@ func TestTrafficSplit(t *testing.T) {
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
 			test.serviceCreate(t, serviceName)
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1"})
-			rev1 := test.latestRevisionOfService(t, serviceName)
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1", "--revision-name", rev1})
+
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
 			tflags := []string{"--tag", fmt.Sprintf("%s=v1,%s=v2", rev1, rev2),
 				"--traffic", "v1=50,v2=50"}
@@ -106,10 +108,12 @@ func TestTrafficSplit(t *testing.T) {
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
 			test.serviceCreate(t, serviceName)
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1"})
-			rev1 := test.latestRevisionOfService(t, serviceName)
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1", "--revision-name", rev1})
+
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
 			tflags := []string{"--traffic", fmt.Sprintf("%s=20,%s=80", rev1, rev2)} // traffic by revision name
 			test.serviceUpdateWithOptions(t, serviceName, tflags)
@@ -122,10 +126,11 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("tag a revision as candidate, without otherwise changing any traffic split",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
+
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1", "--revision-name", rev2})
 
 			tflags := []string{"--tag", fmt.Sprintf("%s=%s", rev1, "candidate")} // no traffic, append new target with tag in traffic block
 			test.serviceUpdateWithOptions(t, serviceName, tflags)
@@ -138,10 +143,11 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("tag a revision as candidate, set 2% traffic adjusting other traffic to accommodate",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
+
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v1", "--revision-name", rev2})
 
 			tflags := []string{"--tag", fmt.Sprintf("%s=%s", rev1, "candidate"),
 				"--traffic", "candidate=2%,@latest=98%"} // traffic by tag name and use % at the end
@@ -156,14 +162,14 @@ func TestTrafficSplit(t *testing.T) {
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
 			// make available 3 revisions for service first
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v3"}) //note that this gives 100% traffic to latest revision (rev3)
-			rev3 := test.latestRevisionOfService(t, serviceName)
+			rev3 := fmt.Sprintf("%s-rev-3", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v3", "--revision-name", rev3}) //note that this gives 100% traffic to latest revision (rev3)
 
 			// make existing state: tag current and candidate exist in traffic block
 			tflags := []string{"--tag", fmt.Sprintf("%s=current,%s=candidate", rev1, rev2)}
@@ -183,8 +189,8 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("update tag from testing to staging for @latest revision",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
 			// make existing state: tag @latest as testing
 			tflags := []string{"--tag", "@latest=testing"}
@@ -202,11 +208,11 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("update tag from testing to staging for a revision (non @latest)",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
 			// make existing state: tag a revision as testing
 			tflags := []string{"--tag", fmt.Sprintf("%s=testing", rev1)}
@@ -226,11 +232,11 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("remove a revision with tag old from traffic block entirely",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
 			// existing state: traffic block having a revision with tag old and some traffic
 			tflags := []string{"--tag", fmt.Sprintf("%s=old", rev1),
@@ -249,8 +255,8 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("tag a revision as stable and current with 50-50% traffic",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
 			// existing state: traffic block having two targets
 			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
@@ -268,17 +274,17 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("revert all traffic to latest ready revision of service",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
-			// existing state: latest revision not getting any traffic
+			// existing state: latest ready revision not getting any traffic
 			tflags := []string{"--traffic", fmt.Sprintf("%s=100", rev1)}
 			test.serviceUpdateWithOptions(t, serviceName, tflags)
 
-			// desired state: revert traffic to latest revision
+			// desired state: revert traffic to latest ready revision
 			tflags = []string{"--traffic", "@latest=100"}
 			test.serviceUpdateWithOptions(t, serviceName, tflags)
 
@@ -291,10 +297,10 @@ func TestTrafficSplit(t *testing.T) {
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
 			// existing state: latest revision has no tag
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			// desired state: tag current to latest ready revision
+			// desired state: tag latest ready revision as 'current'
 			tflags := []string{"--tag", "@latest=current"}
 			test.serviceUpdateWithOptions(t, serviceName, tflags)
 
@@ -306,11 +312,11 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("update tag for a revision as testing and assign all the traffic to it:",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
 			// existing state: two revision exists with traffic share and
 			// each revision has tag and traffic portions
@@ -332,11 +338,11 @@ func TestTrafficSplit(t *testing.T) {
 	t.Run("replace latest tag of a revision with old and give latest to another revision",
 		func(t *testing.T) {
 			serviceName := getServiceNameAndIncrement(serviceBase)
-			test.serviceCreate(t, serviceName)
-			rev1 := test.latestRevisionOfService(t, serviceName)
+			rev1 := fmt.Sprintf("%s-rev-1", serviceName)
+			test.serviceCreateWithOptions(t, serviceName, []string{"--revision-name", rev1})
 
-			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2"})
-			rev2 := test.latestRevisionOfService(t, serviceName)
+			rev2 := fmt.Sprintf("%s-rev-2", serviceName)
+			test.serviceUpdateWithOptions(t, serviceName, []string{"--env", "TARGET=v2", "--revision-name", rev2})
 
 			// existing state: a revision exist with latest tag
 			tflags := []string{"--tag", fmt.Sprintf("%s=latest", rev1)}
@@ -368,9 +374,9 @@ func (test *e2eTest) verifyTargets(t *testing.T, serviceName string, expectedTar
 	assert.DeepEqual(t, expectedTargets, formattedActualTargets)
 }
 
-func (test *e2eTest) latestRevisionOfService(t *testing.T, serviceName string) string {
-	return test.serviceDescribeWithJsonPath(t, serviceName, latestRevisionJsonPath)
-}
+//func (test *e2eTest) latestReadyRevisionOfService(t *testing.T, serviceName string) string {
+//	return test.serviceDescribeWithJsonPath(t, serviceName, LRRJsonPath)
+//}
 
 func (test *e2eTest) serviceDescribeWithJsonPath(t *testing.T, serviceName, jsonpath string) string {
 	command := []string{"service", "describe", serviceName, "-o", jsonpath}

